### DISCUSSION TITLE
Require issue trackers to be enabled for external Python libraries that has communication with external devices or services

### DISCUSSION THREAD

**Discussion by thecode** [home-assistant/architecture#759] [General] [answered]
## Context
**Currently we have the following requirement in our [Development Checklist](https://developers.home-assistant.io/docs/development_checklist):**
_"All communication to external devices or services must be wrapped in an external Python library hosted on [pypi](https://pypi.python.org/pypi)."_

## Proposal

**Require issue trackers to be enabled for external Python libraries that has communication with external devices or services.**

Since Python libraries that communicate with external devices or services are an important part in keeping an integration healthy it is important that issue trackers are not disabled for such packages so that it will be possible to report issues. 

Reporting issues does not guarantee that the repository owner will fix them, however, they allow a discussion there and letting users know that there is an active issue, it may also drive others to try to resolve the issue or contact the device manufacturer or service provider for information needed to resolve it. 

Having issue tracker disabled means the repository owner is not interested in supporting bugs or even simple adjustments to support future Python versions with the package which is not something that can hold for long.

**Comment by balloob** [comment, marked as answer]
Sounds good.

**Reply by bdraco** [reply]
As an alternative, I think we should also allow them to not have issues turned on if they’re a code owner of the integration and it’s only used for Home Assistant as it’s easy to miss issues when you have 100s of repos

**Reply by Jc2k** [reply]
This sounds reasonable. I get a subset of users going to my library to report their issue, instead of filing it with Home Assistant. This means they bypass all the signposting we do in the HA ticket template, other codeowners might not get notified, etc. It's not yet bad enough that I want to turn off issues there, but I can kind of see that I would want to if it increased.

**Reply by bdraco** [reply]
In my experience, there is a significant number of users that don’t understand the difference between reporting it to the library versus reporting it to Home Assistant.  The ones that do report the library are generally much better reports, since they passed the test of finding it in the first place.  When it’s only used in Home Assistant, most the time they have to do the troubleshooting inside of Home Assistant anyways so reporting it to the library doesn’t provide any additional value

**Reply by balloob** [reply]
Instead of turning them off, we could encourage those repositories to use an issue template picker with links to HA, like this: https://github.com/home-assistant-libs/zwave-js-server-python/issues/new/choose

**Reply by bdraco** [reply]
I agree. Thats a much better experience for the user

**Reply by thecode** [reply]
Ok, how does this sound:

Issue trackers must be enabled for external Python libraries that has communication with external devices or services. If the library is only used for Home Assistant and you are a code owner of the integration it encouraged to use an issue template picker with links to [Home Assistant Core](https://github.com/home-assistant/core). For example: [zwave-js-server-python - New Issue](https://github.com/home-assistant-libs/zwave-js-server-python/issues/new/choose)

**Reply by thecode** [reply]
PR to update development checklist: https://github.com/home-assistant/developers.home-assistant/pull/1299

**Reply by MartinHjelmare** [reply]
Change to:

> If the library is mainly used for Home Assistant

We don't control who uses the libraries and they shouldn't know about Home Assistant.
